### PR TITLE
equals.hashcode

### PR DESCRIPTION
**Pull request by IhorYutinDev**

### CODE REVIEW

File: module1/src/test/java/net/broscorp/equals/MyEqualsHashcodeTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  @Test
+  void boxNotEqualTest() {
+    Box box1 = new Box(10, 380);
+    Box box2 = new Box(200, 300);
+
+    assertNotEquals(box1, box2);
+  }
+
```
Comment (pesua) [inline module1/src/test/java/net/broscorp/equals/MyEqualsHashcodeTest.java:43]: Необходимо было проверить все инварианты Equals там пять правил.

**Review by pesua — CHANGES_REQUESTED** [review]

File: module1/src/test/java/net/broscorp/equals/MyEqualsHashcodeTest.java (hunk 1 above)
Comment (IhorYutinDev) [reply module1/src/test/java/net/broscorp/equals/MyEqualsHashcodeTest.java:43]: It is reflexive: for any non-null reference value x, x.equals(x) should return true.
It is symmetric: for any non-null reference values x and y, x.equals(y) should return true if and only if y.equals(x) returns true.
It is transitive: for any non-null reference values x, y, and z, if x.equals(y) returns true and y.equals(z) returns true, then x.equals(z) should return true.
It is consistent: for any non-null reference values x and y, multiple invocations of x.equals(y) consistently return true or consistently return false, provided no information used in equals comparisons on the objects is modified.
For any non-null reference value x, x.equals(null) should return false.

Comment (IhorYutinDev) [reply module1/src/test/java/net/broscorp/equals/MyEqualsHashcodeTest.java:43]: Добавил 5 тестов на проверку вышеуказанных правил

**Comment by Provsky** [comment]
гуд

**Pull request closed without merging**